source/extensions/access_loggers/wasm/config.cc

```diff
@@ -29,22 +29,41 @@ WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& proto_con
   auto vm_id = config.config().vm_config().vm_id();
   auto root_id = config.config().root_id();
   auto configuration = std::make_shared<std::string>(config.config().configuration());
-  auto tls_slot = context.threadLocal().allocateSlot();
+  auto access_log = std::make_shared<WasmAccessLog>(root_id, nullptr, std::move(filter));
+
   // Create a base WASM to verify that the code loads before setting/cloning the for the
   // individual threads.
   auto plugin = std::make_shared<Common::Wasm::Plugin>(
       config.config().name(), config.config().root_id(), config.config().vm_config().vm_id(),
       envoy::api::v2::core::TrafficDirection::UNSPECIFIED, context.localInfo(),
       nullptr /* listener_metadata */, context.scope());
-  auto base_wasm =
-      Common::Wasm::createWasm(config.config().vm_config(), plugin, context.clusterManager(),
-                               context.dispatcher(), context.api());
-  // NB: the Slot set() call doesn't complete inline, so all arguments must outlive this call.
-  tls_slot->set([base_wasm, configuration](Event::Dispatcher& dispatcher) {
-    return std::static_pointer_cast<ThreadLocal::ThreadLocalObject>(
-        Common::Wasm::getOrCreateThreadLocalWasm(*base_wasm, *configuration, dispatcher));
-  });
-  return std::make_shared<WasmAccessLog>(root_id, std::move(tls_slot), std::move(filter));
+
```

Comment:
Can you encapsulate or create a function (e.g. in wasm.h/cc) to do this repeated code?  We have a number of plugin points and are going to have more so it will be repeated several more times.